main_page.php

```diff
@@ -1,11 +1,27 @@
 <?php
-$db = new PDO('mysql:host=db; dbname=collectorapp', 'root', 'password');
-$db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+    /**
+    * Creates a connection to the collectorapp database
+    */
+    function createDBConnection()
+    {
+        $db = new PDO('mysql:host=db; dbname=collectorapp', 'root', 'password');
+        $db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+    }
 
-//Select all the field in my recipes table and fetch-all to create an assoc array
-$query = $db->prepare('SELECT `recipe`, `cuisine`, `time`, `link` FROM `recipes`;');
-$query->execute();
-$recipes = $query->fetchAll();
+    createDBConnection();
+
+    /**
+     * A query to the database that fetches the data in all the fields and creates a recipes assoc array
+    * @return array
+     */
+    function fetchAllRecipes(): array {
+        $query = $db->prepare('SELECT `recipe`, `cuisine`, `time`, `link` FROM `recipes`;');
```

Comment:
How are you getting the connection into the function? It should be passed in as a param